spec/models/validators/Rspec_support_model_helpers_spec.rb

```diff
@@ -0,0 +1,28 @@
+#testing   /spec/support/model_helpers.rb
+require 'rails_helper'
+
+RSpec.describe "Rspec matchers" do
+  let(:test_model) do
+    Class.new do
+      include ActiveModel::Validations
+
+      attr_accessor :date_attr, :other_attr
+
+      validates :date_attr, valid_date_presence: true, allow_nil: true, allow_blank: true
+    end
+  end
+
+  subject { test_model.new }
+
+  describe "should have a functional 'have_validation' matcher" do
+    describe "without options" do
+      it { is_expected.to have_validation :valid_date_presence, :on => :date_attr }
+      it { is_expected.to_not have_validation :valid_date_presence, :on => :other_attr }
+    end
+
+    describe "with options" do
+      it { is_expected.to have_validation :valid_date_presence, :on => :date_attr,
+                                          :options => [:allow_nil, :allow_blank] }
+    end
+  end
+end
```

Comment:
The test failure messages look like this:

`
  Failure/Error: it { is_expected.to have_validation :valid_date_presence, :on => :date_attr }
   expected that #<#<Class:0x0000000a346c00>:0x0000000afba860> would have validation ':valid_date_presence' on ':date_attr' field
`

`
   Failure/Error: it { is_expected.to have_validation :valid_date_presence, :on => :date_attr, :options => [:allow_nil] }
     expected that #<#<Class:0x0000000a0bd9c0>:0x0000000ad55b88> would have validation ':valid_date_presence' on ':date_attr' field with the options [:allow_nil]
`